### DISCUSSION TITLE
Sort collection:previous and collection:next alphabetically

### DISCUSSION THREAD

**Discussion by bruceontheloose** [statamic/cms#5305] [Q&A]
Hello,

I would like the `collection:previous` and `collection:next` links to be sorted alphabetically (per a field such as `title`), rather than by date or manual sort order. The [docs page for `collection:previous`](https://statamic.dev/tags/collection-previous) says it's possible: "If you're on a single entry page and want to show previous  entries in order (publish date, alphabetical, or manual), this is the tag you're looking for." (The [`collection:next`](https://statamic.dev/tags/collection-next) page is similar.) But I can't get it to work. I get:

> Exception  
collection:previous requires ordered or dated collection

Here's the collection's config:
```yaml
title: 'Test Alpha Sort'
template: test_alpha_sort/show
layout: layout
revisions: false
route: '/test/{slug}'
sort_dir: asc
date_behavior:
  past: public
  future: private
```

And here's the relevant template code (I've also tried `sort="title:desc"` and `sort="title"`):
```html
{{ collection:previous in="test_alpha_sort" as="posts" limit="1" sort="title:asc" }}
  {{ if no_results }}
    <p>No more posts to read!</p>
  {{ /if }}
 
  {{ posts }}
    <p class="post">
      <a href="{{ url }}">{{ title }}</a>
    </p>
  {{ /posts }}
{{ /collection:previous }}
```

I know I can make that error go away by turning on **Orderable** and setting **Max Depth** to `1` via the Configure Collection page in the control panel, and then using `sort="order"` in the template, but that's not what I'm after.

How can I sort previous and next alphabetically?

I posted essentially the same question in the old forum in Sept right before it was archived in favor of Discussions here, so that might have been why it went unanswered. Hope you don't mind the repost.

Thanks in advance for any help,

Bruce

**Comment by edalzell** [comment]
Not sure you can sort a collection that has depth.

**Reply by edalzell** [reply]
Yup
<img width="699" alt="CleanShot 2022-02-18 at 13 37 32@2x" src="https://user-images.githubusercontent.com/6069653/154764609-cc1838e5-be5d-42e1-8f0a-7087f3efdd08.png">

**Reply by bruceontheloose** [reply]
Thanks a lot, @edalzell, I had seen that rule. I didn't mention this before, but I also had tried enabling __Oderable__ and setting __Max Depth__ to `1`, and then trying to sort `collection:previous` by `title`, but got the same error. So the error message ("collection:previous requires ordered or dated collection") is a little off in that your collection can be ordered and you can still get the error. (For good measure, I also tried sorting `collection:previous` by `title` with a dated collection: same result.)

Going back to my initial case where the collection doesn't have __Oderable__ enabled or a __Max Depth__ set, sorting by `title` does work as expected with a typical collection tag such as: 

```html
<ol>
{{# This sorts in ascending alphabetical order, and sort="title:desc" reverses it #}}
{{ collection:test_alpha_sort sort="title" }}
  <li>{{ title }}</li>
{{ /collection:test_alpha_sort }}
</ol>

{{# Side note: using sort="title:asc" or no sort parameter gives the same result as sort="title" above. #}}

```

It's just `collection:previous` and `collection:next` that are problematic. So, perhaps the docs are incorrect where they say alpha sorting is possible with them.

**Reply by edalzell** [reply]
sounds like a bug, can you please create an issue?
